### DISCUSSION TITLE
How to get pdb working for micropython?

### DISCUSSION THREAD

**Discussion by pluess** [micropython/micropython#9382] [General]
I have watched https://www.youtube.com/watch?v=RRN_ngx01vM (A Debugger for Micropython?) an wanted to try this myself.

I followed the [Instructions to build Micropython](https://docs.micropython.org/en/latest/develop/gettingstarted.html#compile-and-build-the-code) and got `settrace` working.
But where can I get [pdb](https://docs.python.org/3/library/pdb.html#pdb.Pdb) for Micropython? 

[micropython-pdb](https://pypi.org/project/micropython-pdb/) is just an empty dummy, that does nothing.

I downloaded [the cpython pdb source from github](https://github.com/python/cpython/blob/3.10/Lib/pdb.py) which does not really work:
```
[560] : micropython -m pdb worldtimeapiunittest.py
Traceback (most recent call last):
  File "pdb.py", line 498
SyntaxError: invalid syntax
[561] :
```

So how can i get pdb running for Micropython?

**Comment by jimmo** [comment]
In the video you can see this PR mentioned: https://github.com/micropython/micropython-lib/pull/499
